### DISCUSSION TITLE
Why so many Class components?

### DISCUSSION THREAD

**Discussion by ryskin** [wix/react-native-ui-lib#2351] [General]
I just looked at the source of components and I see many of them are class base, I'm curious. Is this give some benefits or they were created a long time ago?

**Comment by ethanshar** [comment]
Hi @ryskin 
The library is about 5yrs old and most of the components were written when Class component were more popular. 
Today, new components are written as Function components and sometime we refactor old Class components into Functions.

In any case, this is an internal implementation and should not affect you in any way.
Also there's nothing wrong with Class component other than the fact they don't support hooks
